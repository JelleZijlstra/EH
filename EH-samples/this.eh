#!/usr/bin/ehi
# Illustrate the use of $this
class Foo
	private bar: n
		echo 'Private method ' + @string $n
	endfunc
	public baz: n
		echo 'Public method ' + @string $n
		$this.bar: $n
	endfunc
endclass
set foo = new Foo
$foo.baz: 42
