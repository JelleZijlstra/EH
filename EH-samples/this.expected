Public method 42
Private method 42
